carbon-design-system/carbon
# feat(react,carbon-react): add unstable_FlexGrid

Closes #10474 

Add in `FlexGrid` component for a stable flexbox-based Grid implementation. The intention is that teams can keep using `FlexGrid`, `Row`, and `Column` components in v11 before moving to the CSS Grid-based implementation.

#### Changelog

**New**

- Add FlexGrid
- Add CSSGrid
- Add GridSettingsContext and associated setter, getter for the context

**Changed**

- Update Grid to toggle on FlexGrid and CSSGrid based on feature flag
- Update Column to toggle classes based on the current "grid mode"
- Move Grid stories to packages/react/src/components/Grid/next/*

**Removed**

- Remove grid folder in packages/carbon-react

#### Testing / Reviewing

- Verify stable grid implementation works as expected in storybook/locally
- Verify CSS grid implementation works as expected in v11 storybook/locally
- Consider the approach to toggling between behaviors and if there is a simpler option available that we could use instead of authoring new components
